### PR TITLE
Update inventory scripts

### PR DESCRIPTION
**Pull request by AlanCoding**
This is pretty much just pulling the upstream

https://github.com/ansible/ansible/tree/devel/contrib/inventory

I've looked through the changes, nothing looked obvious that it 100% needed a corresponding dependency upgrade with the possible exception of `from boto import sts`. I still intend to do those, but I thought it would be cleanest to do the two tasks separately.

notes in commit message

```
ec2
- added support for tags and instance attributes
- allow filtering RDS instances by tags
- add option to group by platform
- set missing defaults
- make cache unique to script ran
- bug fixes
- implement AND'd filters
azure_rm
- minor python 3 upgrades
cloudforms
- minor regex fix
foreman
- several new configurables
- changes to caching
gce
- python 3 upgrades
- added gce_subnetwork param
openstack
- added `--cloud` parameter
ovirt4
- obtain defaults from env vars
vmware_inventory
- changed imports
- allow for custom filters
- changed host_filters
- error handling
- python 3 upgrades
```

These are only a loose approximation of the changes that happened. The actual commit history has some actions undone by later commits, spelling fixes in comments, etc.

### CODE REVIEW

File: awx/plugins/inventory/ec2.py [hunk 1]
```diff
@@ -605,27 +636,72 @@ def get_instances_by_region(self, region):
...
+
     def get_rds_instances_by_region(self, region):
         ''' Makes an AWS API call to the list of RDS instances in a particular
         region '''
 
+        if not HAS_BOTO3:
+            self.fail_with_error("Working with RDS instances requires boto3 - please install boto3 and try again",
+                                 "getting RDS instances")
+
+        client = ec2_utils.boto3_inventory_conn('client', 'rds', region, **self.credentials)
```
Comment (AlanCoding) [inline awx/plugins/inventory/ec2.py:674]: This might be a concern if the utils used are not present in an old version of Ansible.

Comment (AlanCoding) [reply awx/plugins/inventory/ec2.py:674]: It appears to not be a problem, the code inside of the 2.3 branch:

https://github.com/ansible/ansible/blob/stable-2.3/lib/ansible/module_utils/ec2.py#L120

Comment (AlanCoding) [reply awx/plugins/inventory/ec2.py:674]: and 2.2

**Comment by matburt** [comment]
Might be good to verify some of the important ones to make sure they still work okay

**Review by matburt — APPROVED** [review]

**Comment by matburt** [comment]
We've gotten in trouble before where there were major regressions in the ec2, openstack, and vmware ones specifically.

**Comment by AlanCoding** [comment]
I tested ec2 and openstack, I had those on hand. They're working same as before. Granted, these are very small and trivial examples.

**Pull request merged**